mlflow/pyspark/ml/__init__.py

```diff
@@ -0,0 +1,285 @@
+        if len(_SparkTrainingSession._session_stack) > 0:
+            self._parent = _SparkTrainingSession._session_stack[-1]
+            self.allow_children = (
+                _SparkTrainingSession._session_stack[-1].allow_children and self.allow_children
+            )
+        _SparkTrainingSession._session_stack.append(self)
+        return self
+
+    def __exit__(self, tp, val, traceback):
+        _SparkTrainingSession._session_stack.pop()
+
+    def should_log(self):
+        """
+        Returns True when at least one of the following conditions satisfies:
+
+        1. This session is the root session.
+        2. The parent session allows autologging and its class differs from this session's class.
+        """
+        return (self._parent is None) or (
+            self._parent.allow_children and self._parent.clazz != self.clazz
+        )
+
+
+def _get_warning_msg_for_fit_call_with_a_list_of_params(estimator):
+    return (
+        "Skip pyspark ML autologging when calling "
+        + f"{_get_fully_qualified_class_name(estimator)}.fit with a list of params,"
+        + "if you want to autolog for this case, you convert code to call `fit` with "
+        + "each single param map."
+    )
+
+
+@experimental
+@autologging_integration(FLAVOR_NAME)
+def autolog(
+    log_models=True,
+    disable=False,
+    exclusive=False,
+    disable_for_unsupported_versions=False,
+    silent=False,
+):  # pylint: disable=unused-argument
+    """
+    Enables (or disables) and configures autologging for pyspark ml estimators.
+    This method is not threadsafe and assumes a
+    `SparkSession
+    <https://spark.apache.org/docs/latest/api/python/pyspark.sql.html#pyspark.sql.SparkSession>`_
+    already exists.
+    This API requires Spark 3.0 or above.
+
+    **When is autologging performed?**
+      Autologging is performed when you call ``Estimator.fit``
+
+    **Logged information**
+      **Parameters**
+        - Parameters obtained by ``estimator.params``. If a param value is also an ``Estimator``,
+          then params in the the wrapped estimator will also be logged, the nested param key
+          will be `{estimator_uid}.{param_name}`
+
+      **Tags**
+        - An estimator class name (e.g. "LinearRegression").
+        - A fully qualified estimator class name
+          (e.g. "pyspark.ml.regression.LinearRegression").
+
+      **Artifacts**
+        - An MLflow Model with the :py:mod:`mlflow.spark` flavor containing a fitted estimator
+          (logged by :py:func:`mlflow.spark.log_model()`). Note that large models may not be
+          autologged for performance and storage space considerations.
+          See ``log_models`` param below for details.
+
+
+    :param log_models: If ``True``, if trained models are in allowlist, logged as MLflow model
+                       artifacts. If ``False``, trained models are not logged.
+                       Note: the built-in allowlist excludes some models which model may be large,
```

Comment:
```suggestion
                       Note: the built-in allowlist excludes some models which model may be large.
```

Reply:
Is there a way the user can view the default allowlist / can we document the default allowlist here?

Reply:
I think we can use the `literalinclude` directive to insert the allowlist in the document: 

https://stackoverflow.com/a/41592881/6943581